src/concepts/imperative.md

```diff
@@ -0,0 +1,167 @@
+# Imperative Programming
+
+A fun thing we can do with functional programming in SML is imperative
+programming! Does it seem counterintuitive? Perhaps. Is it cool? OF COURSE!
+Background knowledge with languages like C, C++, Java, and Python will be
+helpful but is not required.
+
+## Reference Cells
+
+The way SML goes about implementing imperative features is by using something
+called **_reference types_**. These are **_references_** to a value of a
+certain type. This is very similar to how a pointer might point to a certain
+place in memory. More formally:
+
+> A value of type `t ref` is a mutable cell capable of holding a value of type
+> `t`. For example, an `int ref` is a reference to an `int`.
+
+_Reference Cells, Ref Cells, and Mutable Cells all refer to the same thing in_
+_this article._
+
+## Functions / Expressions to Know
+
+There's a few different expressions and functions we can leverage to help us
+interface with these **_reference types_**.
+
+### Ref Cells `ref`
+
+First, we need to make something of this reference type. In other words, we
+need a way to initialize our pointers. `ref` is also a function that helps us
```

Comment:
> `ref` is also a function

It would probably be more accurate to point out that `ref` is a constructor.

Reply:
I see what you mean. Doesn't this imply that you're able to pattern match on `ref`?

 _(Made the change but going to leave this unresolved)_

Reply:
Yes. You can pattern match on `ref`.